pkg/server/structlogging/structlogging_test: TestHotRangesStats failed

pkg/server/structlogging/structlogging_test.TestHotRangesStats [failed](https://tanzanite.cluster.engflow.com/invocations/default/6bd1c1e1-cd79-4b7d-bec9-d6493bdda7dc?testReportRun=2&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyL3N0cnVjdGxvZ2dpbmc6c3RydWN0bG9nZ2luZ190ZXN0) on release-24.2 @ [c3712a42b57bcd6dd4285fc52d915fd4dc931e6e](https://github.com/cockroachdb/cockroach/commits/c3712a42b57bcd6dd4285fc52d915fd4dc931e6e):


```
=== RUN   TestHotRangesStats
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestHotRangesStats2837189451
    hot_ranges_log_test.go:158: Logged ranges should be unique per node for this test.
        found range on node 2 and node 2: ‹test› ‹public› ‹foo› ‹foo_pkey› 73
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestHotRangesStats2837189451
--- FAIL: TestHotRangesStats (76.55s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestHotRangesStats.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-42885

pkg/server/structlogging/structlogging_test.TestHotRangesStats [failed](https://tanzanite.cluster.engflow.com/invocations/default/9bd48734-ee8d-4513-80c7-3287a4d6f767?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyL3N0cnVjdGxvZ2dpbmc6c3RydWN0bG9nZ2luZ190ZXN0) on release-24.2 @ [5764366875adca70c3a3ff3d4d13aaf599bc53a7](https://github.com/cockroachdb/cockroach/commits/5764366875adca70c3a3ff3d4d13aaf599bc53a7):


```
=== RUN   TestHotRangesStats
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestHotRangesStats2711829379
    hot_ranges_log_test.go:158: Logged ranges should be unique per node for this test.
        found range on node 2 and node 2: ‹test› ‹public› ‹foo› ‹foo_pkey› 71
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestHotRangesStats2711829379
--- FAIL: TestHotRangesStats (36.72s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #134144 pkg/server/structlogging/structlogging_test: TestHotRangesStats failed [C-test-failure O-robot T-observability branch-release-24.2.5-rc release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestHotRangesStats.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/server/structlogging/BUILD.bazel
+++ b/pkg/server/structlogging/BUILD.bazel
@@ -41,6 +41,7 @@ go_test(
         "//pkg/sql/catalog/desctestutils",
         "//pkg/testutils",
         "//pkg/testutils/serverutils",
+        "//pkg/testutils/skip",
         "//pkg/testutils/sqlutils",
         "//pkg/testutils/testcluster",
         "//pkg/util/leaktest",

--- a/pkg/server/structlogging/hot_ranges_log_test.go
+++ b/pkg/server/structlogging/hot_ranges_log_test.go
@@ -20,6 +20,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/desctestutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
@@ -79,6 +80,8 @@ func TestHotRangesStats(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	sc := log.ScopeWithoutShowLogs(t)
 	defer sc.Close(t)
+
+	skip.UnderRace(t)
 
 	ctx := context.Background()
 	spy := hotRangesLogSpy{t: t}

